tower-http/src/add_extension.rs

```diff
@@ -52,6 +52,13 @@ impl<S, T> AddExtension<S, T> {
     }
 
     define_inner_service_accessors!();
+
+    /// Returns a new [`Layer`] that wraps services with a `AddExtensionLayer` middleware.
```

Comment:
nit: the middleware is `AddExtension`, not the layer:
```suggestion
    /// Returns a new [`Layer`] that wraps services with a `AddExtension` middleware.
```